tedana/metrics/kundu_fit.py

```diff
@@ -126,16 +123,17 @@ def dependence_metrics(catd, tsoc, mmix, t2s, tes, ref_img,
     # compute Betas and means over TEs for TE-dependence analysis
     betas = get_coeffs(utils.unmask(catd, mask),
                        mmix,
-                       np.repeat(mask[:, np.newaxis], len(tes), axis=1))
+                       np.repeat(mask[:, np.newaxis], len(tes), axis=1),
+                       add_const=True)
     betas = betas[mask, ...]
     n_voxels, n_echos, n_components = betas.shape
     mu = catd.mean(axis=-1, dtype=float)
     tes = np.reshape(tes, (n_echos, 1))
     fmin, _, _ = getfbounds(n_echos)
 
-    # set up Xmats
-    X1 = mu.T  # Model 1
-    X2 = np.tile(tes, (1, n_voxels)) * mu.T / t2s.T  # Model 2
+    # set up design matrices
+    X1 = mu.T  # Model 1: TE-independence model
+    X2 = np.tile(tes, (1, n_voxels)) * mu.T  # Model 2: TE-dependence model
```

Comment:
In the current code, X2 = np.tile(tes, (1, n_voxels)) * mu.T / t2s.T
The t2s is now excluded from this calculation and t2S is no longer even passed as a variable to dependence metrics. I'll need to look a bit more carefully at what is happening in the new lines 158-183 (That section could use a few more comments), but it doesn't seem to have a replacement from the removable of t2s here. Am I missing something?

Reply:
That is correct. The thing is, `t2s` has no impact on any calculated values, so it doesn't need to be passed into this function.

Reply:
I'm either missing something here or need to dig into what happens next to see if t2s just cancels itself out. In the current code `t2s` is used to calculate `X2` This is removed from this pull request.
`X2` is used to calculated `coeffs_R2` and `pred_R2` which ends up in a bunch of other variables including `F_R2_maps` On the face, that seems fairly significant and removing the division by `t2s` in the `X2` calculation is essentially removing all T2* information from the fit to the R2 model.

Reply:
`t2s` is an (S) array, while `X2` is an (S x E) array. Essentially, `t2s` acts as a scalar for the design matrix of E values for each voxel.  Scaling the design matrix will impact `coeffs_R2` and `SSE_R2` (though not `pred_R2`), but those cancel each other out. Since it's just a scalar for each voxel's equation and F-statistics are calculated independently across voxels, it also shouldn't impact goodness of fit for the model.

If you run it both ways (with vs. without `t2s`), the Kappa values end up the same.

Reply:
OK. Took a while to warp my head around this & then run on a data set to confirm, but this makes sense now. I'd say we should comment this code more, but it's going to be replaced by the clearer and better commented #475 anyway.